faqrenderer/faq.md

```diff
@@ -0,0 +1,121 @@
+
+## How do I restart my JupyterLab server?
+
+On Libretexts JupyterHub, you spawn servers that provide you with JupyterLab software and you execute your code using kernels. If you are actively running code and something unexpected occurs, [restarting the kernel](https://jupyterlab.readthedocs.io/en/stable/user/running.html) can often be an easy solution. However, sometimes the problem may be deeper than that, especially if some of your files were incorrectly modified. In this case, it could be a good idea to restart your server. 
+
+Go to File->Hub Control Panel and you will be brought to a page with "Stop Server" and "My Server" buttons. Press Stop Server and you should see that that the My Server button will dim; this means that the server is stopping. When the server has stopped (this should take less than 15 seconds), the webpage will prompt you with a "Start My Server" option. Press it, launch the server, and then you will be brought back to the Server Options page. Proceed with the environment of your choice, and you will now have a fresh server.
+
+By default, your server will always shutdown after 1 hour of inactivity. All files which you wish to modify and save across restarts must be located in your `/home/jovyan` directory; everything else will be wiped. Think of these user files as being stored in a cloud, and each time you start JupyterLab, we provide you with a brand new server which has downloaded your specific files. For more technical information on how your files are stored, see the Jupyterhub for Kubernetes documentation [here](https://zero-to-jupyterhub.readthedocs.io/en/latest/customizing/user-environment.html#about-user-storage-and-adding-files-to-it). 
+
+## Can I recieve accouncements for downtime and other critical information about jupyter.libretexts.org?
+
+If you wish to recieve updates from the JupyterTeam about possible downtime for jupyter.libretexts.org, you can subscribe to our [mailing list](https://lists.ucdavis.edu/sympa/subscribe/flock-announce). Just enter your email and you will recieve emails whenever we notify Hub users about downtime and maintenance.
+
+# For Instructors
+
+## How do I distribute files to students?
+
+You can distribute files using [nbgitpuller](https://jupyterhub.github.io/nbgitpuller/index.html). Please store your files in a GitHub repository and we can set up a custom environment that will update the files whenever a student spawns a server. This is a good option since students don't need to understand git, but keep in mind that nbgitpuller does have [some limitations](https://jupyterhub.github.io/nbgitpuller/topic/automatic-merging.html). Other options include hosting your files on an external site, like GitHub or Google Drive, where students can download them.
```

Comment:
Do we know that we have this feature installed and enabled? Have you tried it?

Reply:
I've never tried it and I don't know if it works. We could test, but based on your comments below, it looks like we're just removing this.

Reply:
I didn't mean to remove this section. Please see if nbgitpuller is in installed and works. We'd like to support it.

Reply:
We've added nbgitpuller to our repo2docker environment and have confirmed that it works. Basically nbgitpuller allows instructors to make links tied to any git repository they would like, and then whenever students click the link, it adds the files within that repo to their jupyterlab home directory (which can be specified within the link). The FAQ here makes it sound like we need to set up some custom environments, but I don't think that this would be necessary for nbgitpuller.

[Here](https://jupyterhub.github.io/nbgitpuller/link?hub=https://jupyter.libretexts.org&app=jupyterlab) is a nbgitpuller link generator that is prepopulated with all the necessary fields except for the git repo to pull from. We can briefly direct instructors to this link and then tell them to fill out the git repo field, and then share the link with their students. Or you may want us to do this process for them as the current wording implies. Let us know what you think. @moorepants

Reply:
If the instructor can setup their use of nbgitpuller independent of us, then we should instruct them to do so. It isn't clear to me what we'd need to do to help them get this setup.